### ISSUE TITLE
Saml SSO configuration yaml file referenced in the docs not working properly with newer versions of spring boot

### ISSUE THREAD

**Issue by Anubhav-2000** [spring-projects/spring-security#12810]
I was using the spring security docs and this link as a reference to implement SSO: https://medium.com/digital-software-architecture/spring-boot-spring-security-with-saml-2-83d87df5b470

This seems to work if the spring boot starter parent dependency is version 2.4.2. But fails with version 2.7.2. It gives an error Invalid signature for object [id…]

I messed around with the code a bit and it only worked when i changed the identityprovider tag in the yaml file to assertingparty tag.

So the spring docs says to use this yaml file: https://docs.spring.io/spring-security/reference/servlet/saml2/login/overview.html#saml2-specifying-identity-provider-metadata

But the correct yaml file is:
`spring:
  security:
    saml2:
      relyingparty:
        registration:
          adfs:
            assertingparty:
              entity-id: https://idp.example.com/issuer
              verification.credentials:
                - certificate-location: "classpath:idp.crt"
              singlesignon.url: https://idp.example.com/issuer/sso
              singlesignon.sign-request: false`

**Comment by jvalkeal** [comment]
You're right that it's expected to work with `assertingparty` as Spring Boot 2.7.x deprecated `identityprovider` in favour of `assertingparty`. We need to change docs to mention something about this deprecation.

Can you share a bit more about the error, do you possible have a sample we could run? Dis it fail at server startup or at runtime?

**Comment by jvalkeal** [comment]
Also looking https://github.com/spring-projects/spring-boot/issues/30128, some new properties are now handled via boot's auto-config, but those added in that issue don't check deprecated properties.

I also saw a simple boot app(2.7.x) not to fail at startup when using `identityprovider` but it did with `assertingparty`. I expected behaviour to be same.

Also boot 3.0.x did remove deprecations so `identityprovider` is not processed anymore.

**Comment by Anubhav-2000** [comment]
Hi @jvalkeal , 
When i have my yaml file like this:  https://docs.spring.io/spring-security/reference/servlet/saml2/login/overview.html#saml2-specifying-identity-provider-metadata

Along with my security config file as:
```
@EnableWebSecurity
public class SecurityConfig {
    @Autowired
    private RelyingPartyRegistrationRepository relyingPartyRegistrationRepository;

    @Bean
    SecurityFilterChain app(HttpSecurity http) throws Exception {
        // @formatter:off
        RelyingPartyRegistrationResolver relyingPartyRegistrationResolver = new DefaultRelyingPartyRegistrationResolver(this.relyingPartyRegistrationRepository);
        Saml2MetadataFilter filter = new Saml2MetadataFilter(relyingPartyRegistrationResolver, new OpenSamlMetadataResolver());
        http
                .authorizeHttpRequests((authorize) -> authorize
                        .anyRequest().authenticated()
                )
                .saml2Login(withDefaults())
                .addFilterBefore(filter, Saml2WebSsoAuthenticationFilter.class);;
        // @formatter:on

        return http.build();
    }
}
```


I get this error:
<img width="1218" alt="Screenshot 2023-03-08 at 11 57 02 PM" src="https://user-images.githubusercontent.com/69172043/223803030-15224ba9-7452-4b9a-b3ff-23b6fa4eaaef.png">

In the logs i get:
Signature of Assertion 'id170959100448639611173959713' from Issuer: (url of issuer) not valid

My pom.xml:
```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.7.9</version>
		<relativePath/> <!-- lookup parent from repository -->
	</parent>
	<groupId>com.example</groupId>
	<artifactId>demo</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<name>demo</name>
	<description>Demo project for Spring Boot</description>
	<properties>
		<java.version>17</java.version>
	</properties>
	<dependencies>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-web</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.security</groupId>
			<artifactId>spring-security-saml2-service-provider</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-thymeleaf</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-actuator</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-security</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.security</groupId>
			<artifactId>spring-security-test</artifactId>
			<scope>test</scope>
		</dependency>
	</dependencies>

	<build>
		<plugins>
			<plugin>
				<groupId>org.springframework.boot</groupId>
				<artifactId>spring-boot-maven-plugin</artifactId>
			</plugin>
		</plugins>
	</build>

</project>
```

**Comment by jvalkeal** [comment]
We'll use this issue to just update docs.

**Comment by Anubhav-2000** [comment]
Should i still update the docs to reflect assertingparty instead of identityprovider?

**Comment by jvalkeal** [comment]
@Anubhav-2000 If you want to create a PR, then it should be done against branch `6.0.x`. Thx for your help.

**Comment by gintophilip** [comment]
I was trying to setup a minimal SSO based on SAML with Azure Entra ID. Based on the guide provided at [saml guide](https://docs.spring.io/spring-security/reference/servlet/saml2/login/overview.html).
Ran into some errors and finally reached here. Based on this i changed the **identityprovider** to **relyingparty** . But it did not work.
```
spring:
  application:
    name: demosamlazure
  security:
    saml2:
      relyingparty:
        registration:
          adfs:
            assertingparty:
              entity-id: demo-saml-app
              singlesignon.url: {single sign on url}
              singlesignon.sign-request: false
```
This is my config. 
The entity Id is obtained from azure Entra Id
![image](https://github.com/spring-projects/spring-security/assets/166334082/ee2b2b3a-d7dc-4435-8522-386312402c96)

I expected this to work. Since we provided both relying party and asserting party configs. 
```
spring:
  application:
    name: demosamlazure
  security:
    saml2:
      relyingparty:
        registration:
          azure-saml:
            assertingparty:
              entity-id: demo-saml-app
              singlesignon.url: {single sign on url}
              singlesignon.sign-request: false
```
But faced with following error .
![image](https://github.com/spring-projects/spring-security/assets/166334082/e81e5969-2ffd-4f10-ba66-02eca43e89b5)

At this point I'm confused about what does adfs means. After some tinkering i gave a random value **azure-saml** instead of adfs based on the documentation.

> The registrationId is an arbitrary value that you choose for differentiating between registrations.  

This does not help either . The error I received is the same as above . instead of /adfs it was /azure-saml
`AADSTS700016: Application with identifier 'http://localhost:8080/saml2/service-provider-metadata/azure-saml' was not found in the directory 'Default Directory'. This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You may have sent your authentication request to the wrong tenant.`

What am I doing wrong here?  Can some one help?

**Comment by danilopiazza** [comment]
I stumbled onto the same issue, and I updated the documentation (against `main`) in PR #17081.
